docs/src/pages/documentation/getting-started.md

```diff
@@ -50,80 +49,158 @@ Identify a place in your existing application's HTML or templates where you'd li
+Here's where things get interesting. Web Fragments uses a lightweight middleware - _the fragment gateway_ - to coordinate all the moving parts. The gateway sits in front of your application and routes requests to the right destination: your existing app, or one of your fragments.
+
+Why do we need this? The gateway enables you to operate your entire micro-frontend application on a single origin, which brings huge benefits:
+
+- **No CORS headaches**: Everything shares the same origin
+- **Better performance**: No cross-origin preflight requests
+- **Simpler security**: The [same-origin policy](https://web.dev/articles/same-origin-policy) works in your favor
+- **Unified deployment**: One domain, multiple independent services behind it
 
-For the gateway to understand how to route your web fragment's requests, the fragment needs to be registered.
+Think of the gateway as a smart router that inspects each incoming request and says, "This request is for the shell app," or "This request is for the party-button fragment," and routes accordingly.
+
+For the gateway to make these routing decisions, you need to tell it about your fragments:
 
 ```js
 import { FragmentGateway } from "web-fragments/gateway";
 
-// initialize the gateway
+// Initialize the gateway
 const myGateway = new FragmentGateway();
 
-// register our fragment
+// Register our fragment
 myGateway.registerFragment({
-	// a unique ID of the fragment
+	// A unique ID matching the fragment-id attribute in your HTML
 	fragmentId: "party-button",
-	piercingClassNames: [],
+	// The endpoint where your fragment is hosted
+	// This can be an HTTP URL, or a Service Binding on Cloudflare
 	endpoint: "https://party-button.demos.web-fragments.dev",
+
+	// Route patterns tell the gateway which URLs belong to this fragment
 	routePatterns: [
-		// url pattern for fetching all assets of this fragment, this pattern is determined by the fragment and should be unique:
-		"/__wf/dev.web-fragments.demos.party-button/:_*",
-		// routable url in the final application where this fragment will be initialized (adjust as needed per step 2c):
+		// Assets pattern: uniquely identifies this fragment's static files
+		// The :_* is a wildcard that matches everything after this point
+		"/__wf/party-button.demos.web-fragments.dev/:_*",
+
+		// Application pattern: where in your app this fragment appears
+		// Adjust this to match where you placed the <web-fragment> element
 		"/",
 	],
+
+	// Optional: CSS classes for positioning during server-side piercing
+	piercingClassNames: [],
 });
 ```
 
-<!--
-Apart from registering the custom elements, `fragments` must be registered in the [fragment gateway](./gateway). In order to do so, the gateway must be imported to the server application.
+Let's break down the `routePatterns`:
+
+- The **asset pattern** (`/__wf/.../:_*`) captures requests for the fragment's JavaScript, CSS, images, etc.
```

Comment:
it feels like this is the third time you explain the same. can you please dedupe?